src/tfr_navigation/src/navigation_client.cpp

```diff
@@ -0,0 +1,62 @@
+#include <navigation_client.h>
+        
+/**
+ *  default constructor, brings up a server to listen to a specific action
+ * */
+NavigationClient::NavigationClient(std::string action_name):client{action_name, true}
+{
+    client.waitForServer();
+}
+
+
+/**
+ * Nonblocking call to Navigate to an arbirtrary defined place as defined in the Navigation
+ * Goal constants header.
+ *
+ * action - The byte code corresponding to a location in the world map
+ * */
+void NavigationClient::navigate(uint8_t location)
+{
+    tfr_msgs::NavigationGoal goal;
+    goal.location_code= location;
+    client.sendGoal(goal,
+            boost::bind(&NavigationClient::finished,this, _1,_2),
+            Client::SimpleActiveCallback(),
```

Comment:
What does the `SimpleActiveCallback()` function do?  I see it's a member of the `SimpleActionClient` class provided by ROS.  Do we need a function for start callbacks?

Reply:
It's a dummy function supplied by the library implementers its the equivalent of null.

Reply:
Might it be better to use `NULL`, then? It's a bit more clear as to what's happening.

Reply:
Well I got this example from the tutorials on ros. I looked at the source code and it looks like this actually does something for that method a little more complicated than that, so I don't know if changing that type will be free of consequences.

Source: http://docs.ros.org/jade/api/actionlib/html/simple__action__client_8h_source.html

So how about I document with a comment explaining whats going on and leave it as is?